### PR TITLE
rabbitmq-plugins list: allow warnings to be suppressed

### PR DESCRIPTION
**Pull request by Ayanda-D**
## Proposed Changes

Some tools list plugins e.g. in `json` format, and can fail to decode output from missing plugins warning banner. We want to be able to optionally list plugins with no warning banner - avoiding any extra work on calling tools (e.g. in the case of sharing config files across new and legacy brokers). Example, with `rabbitmq_unknown` in enabled_plugins file:

#### with warning
```
ayanda@host$ rabbitmq-plugins list --formatter json
WARNING - plugins currently enabled but missing: rabbitmq_unknown

{"status":"running","format":"normal","plugins":[{"enabled":"not_enabled","name":"rabbitmq_amqp1_0","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_cache","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_http","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_ldap","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_oauth2","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_mechanism_ssl","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_consistent_hash_exchange","running":false,"version":[51,46,49
```

#### with no warning

```
ayanda@host$ rabbitmq-plugins list --formatter json --ignore-warnings
{"status":"running","format":"normal","plugins":[{"enabled":"not_enabled","name":"rabbitmq_amqp1_0","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_cache","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_http","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_ldap","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_backend_oauth2","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_auth_mechanism_ssl","running":false,"version":[51,46,49,50,46,56],"running_version":null},{"enabled":"not_enabled","name":"rabbitmq_consistent_hash_exchange","running":false,"version":[51,46,49
```

**NOTE**: --silent and --quiet options more for generic CommandBehaviour messages

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc.

### CODE REVIEW

File: deps/rabbitmq_cli/lib/rabbitmq/cli/plugins/commands/list_command.ex [hunk 1]
```diff
@@ -20,9 +20,15 @@ defmodule RabbitMQ.CLI.Plugins.Commands.ListCommand do
   def merge_defaults(args, opts), do: {args, Map.merge(default_opts(), opts)}
 
   def switches(),
-    do: [verbose: :boolean, minimal: :boolean, enabled: :boolean, implicitly_enabled: :boolean]
+    do: [
+      verbose: :boolean,
+      minimal: :boolean,
+      enabled: :boolean,
+      implicitly_enabled: :boolean,
+      no_warn: :boolean
```
Comment (michaelklishin) [inline deps/rabbitmq_cli/lib/rabbitmq/cli/plugins/commands/list_command.ex:28]: Can we rename this to `ignore_warnings`?

Comment (Ayanda-D) [reply deps/rabbitmq_cli/lib/rabbitmq/cli/plugins/commands/list_command.ex]: done! thanks was hoping for a better name suggestion 😹 👍

**Review by michaelklishin — CHANGES_REQUESTED** [review]
I have only now noticed that the example uses `--formatter=json`.

For such cases, there is already `--silent`, `--quiet` and `RabbitMQ.CLI.Core.Config.output_less/1`.

Instead of a new option, we can simply use `RabbitMQ.CLI.Core.Config.output_less/1` where `ignore_warnings` is used in this PR to ignore the warnings.

**Comment by michaelklishin** [comment]
See https://github.com/rabbitmq/rabbitmq-server/pull/10870.

**Comment by Ayanda-D** [comment]
nice, reusing `-s` and `-q` better approach. I'd traced `opts` expecting --silent and --quiet to always be there, defaulting to false. didn't realise we already have `RabbitMQ.CLI.Core.Config.output_less?/1` defaulting to `false` for both regardless - good to know/learn for next time! 🙌

@michaelklishin Thank you!

**Pull request closed without merging**